community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java

```diff
@@ -1199,18 +1206,22 @@ private ByteBuffer allocate( int sizeIndex )
 
         void free()
         {
+            assertNotFreed();
+
             try
             {
                 clear();
             }
             finally
             {
                 buf = null;
+                freeCall = new Exception();
             }
         }
 
         synchronized void put( int pos, byte[] bytes, int offset, int length )
         {
+            assertNotFreed();
             verifySize( pos + length );
```

Comment:
You can change these explicit asserts into `buf()` calls that stores the result to a local variable.